### PR TITLE
[WIP] [SPARK-389] teragen hdfs integration test

### PR DESCRIPTION
**Pull request by mgummelt**

### CODE REVIEW

**Comment by mgummelt** [comment]
retest this please

**Comment by mgummelt** [comment]
cc @susanxhuynh

File: tests/test.py [hunk 1]
```diff
@@ -7,11 +7,56 @@
...
+                              "principal": "service-acct",
+                              "secret_name": "secret" }
+
+    shakedown.install_package('spark', options_json=options, wait_for_completion=True)
+
+    def pred():
+        dcos_url = dcos.config.get_config_val("core.dcos_url")
+        spark_url = urllib.parse.urljoin(dcos_url, "/service/spark")
+        status_code = dcos.http.get(spark_url).status_code
+        return status_code == 200
```
Comment (susanxhuynh) [inline tests/test.py]: What is this for? Is it to replace the "sleep 30 s." above?

Comment (susanxhuynh) [inline tests/test.py]: I wonder if you need the same check for hdfs.

Comment (mgummelt) [inline tests/test.py]: It's to wait for adminrouter to pick up spark and serve it at (/service/spark).  We don't use adminrouter to access HDFS

Comment (susanxhuynh) [inline tests/test.py]: OK. Should we add in a check to make sure all the HDFS servers (JournalNodes, NameNodes, ZKFCs, DataNodes) are up before launching the Spark job? We do that in the HDFS integration tests.

Comment (mgummelt) [inline tests/test.py]: Yea I think that's why I'm seeing failures right now.  Will report back once I diagnose.

**Comment by mgummelt** [comment]
blocked on https://github.com/dcos/shakedown/pull/110

**Comment by mgummelt** [comment]
restest this please

**Pull request merged**